src/store/actions.js

```diff
@@ -0,0 +1,186 @@
+      dispatch({
+        type: actionTypes.SIGN_UP,
+        payload: { sessionData, isLoggedIn: true },
+      });
+    } catch (err) {
+      dispatch({
+        type: actionTypes.SIGN_UP,
+        payload: { sessionData: { ...initSession }, isLoggedIn: false },
+      });
+    }
+  })();
+};
+const signInAction = (dispatch, { email, password }) => {
+  (async () => {
+    try {
+      const {
+        data: {
+          encodedToken,
+          foundUser: { email: emailData, createdAt, updatedAt, cart, wishlist },
+        },
+      } = await axios.post("/api/auth/login", {
+        email,
+        password,
+      });
+
+      const sessionData = {
+        emailData,
+        encodedToken,
+        createdAt,
+        updatedAt,
+      };
+      localStorage.setItem("authSession", JSON.stringify(encodedToken));
+      dispatch({
+        type: actionTypes.SIGN_IN,
+        payload: { sessionData, isLoggedIn: true, cart, wishlist },
+      });
+    } catch (err) {
+      dispatch({
+        type: actionTypes.SIGN_IN,
+        payload: { sessionData: { ...initSession }, isLoggedIn: false },
+      });
+    }
+  })();
+};
+
+const logoutAction = (dispatch) => {
+  localStorage.removeItem("authSession");
+  dispatch({ type: actionTypes.LOGOUT });
+};
+
+const addToWishlistAction = (dispatch, product) => {
+  (async () => {
+    try {
+      const {
+        data: { wishlist },
+      } = await axios.post(
+        WISHLIST_URL,
+        { product },
+        {
+          headers: {
+            authorization: localStorage.getItem("authSession"),
+          },
+        }
+      );
+      console.log(wishlist);
+      dispatch({
+        type: actionTypes.ADD_TO_WISHLIST,
+        payload: wishlist,
+      });
+    } catch (err) {
+      console.log(err);
+      dispatch({
+        type: actionTypes.ADD_TO_WISHLIST,
+        payload: [],
+      });
+    }
+  })();
+};
+
+const removeFromWishlistAction = (dispatch, product) => {
+  const { id } = product;
+  (async () => {
+    try {
+      const data = await axios.delete(`${WISHLIST_URL}/${id}`, {
+        headers: {
+          authorization: localStorage.getItem("authSession"),
+        },
+      });
+      console.log(data);
+      dispatch({
+        type: actionTypes.REMOVE_FROM_WISHLIST,
+        payload: id,
+      });
+    } catch (err) {
+      dispatch({
+        type: actionTypes.REMOVE_FROM_WISHLIST,
+        payload: null,
+      });
+    }
+  })();
+};
+
+const addToCartAction = (dispatch, product) => {
+  (async () => {
+    try {
+      const {
+        data: { cart },
+      } = await axios.post(
+        `${CART_URL}`,
+        { product },
+        {
+          headers: {
+            authorization: localStorage.getItem("authSession"),
+          },
+        }
+      );
+      console.log(cart);
+      dispatch({
+        type: actionTypes.ADD_TO_CART,
+        payload: cart,
+      });
+    } catch (err) {
```

Comment:
if a user isnt logged in , it should take the user to the login page, not do this add to cart empty payload? 
clicking on add to cart button seems broken unless logged in. just add a navigate link to your login page